crates/transaction-pool/src/traits.rs

```diff
@@ -717,9 +718,9 @@ pub enum PoolUpdateKind {
 ///
 /// This is used to update the pool state accordingly.
 #[derive(Clone, Debug)]
-pub struct CanonicalStateUpdate<'a> {
+pub struct CanonicalStateUpdate<'a, B = reth_primitives::Block> {
     /// Hash of the tip block.
-    pub new_tip: &'a SealedBlock,
+    pub new_tip: &'a B,
```

Comment:
`SealedBlockFor<B>` will work here if you add trait bound `B: Block`, where `Block` is imported from `reth-primitives-traits`